analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,401 @@
+head(epic_df)
+```
+
+Below is a table with high-level diagnoses for each ScPCA project:
+
+
+| Project     | High-level diagnosis |
+|-------------|----------------------|
+| SCPCP000001 | High-grade glioma    |
+| SCPCP000002 | Low-grade glioma     |
+| SCPCP000006 | Wilms tumor          |
+| SCPCP000009 | CNS tumors           |
+| SCPCP000017 | Osteosarcoma         |
+
+For ease interpreting subsequent figures, we'll add these high-level diagnoses into the data frame with a project number indicator (excluding leading 0s).
+
+```{r}
+epic_df <- epic_df |>
+  dplyr::mutate(
+    project_diagnosis = dplyr::case_match(
+      project_id, 
+      "SCPCP000001" ~ "01: High-grade glioma", 
+      "SCPCP000002" ~ "02: Low-grade glioma", 
+      "SCPCP000006" ~ "06: Wilms tumor", 
+      "SCPCP000009" ~ "09: CNS tumors", 
+      "SCPCP000017" ~ "17: Osteosarcoma"
+    ))
+
+```
+
+## Visualization
+
+### Proportions of cell types
+
+Considering all categories of cells, what proportions of cell types are represented across samples?
+
+
+```{r fig.height=8, fig.width=16}
+# First, we'll turn the otherCells level into `NA` for ease in coloring
+epic_otherna_df <- epic_df |>
+  dplyr::mutate(epic_celltype = ifelse(
+    epic_celltype == "otherCells", 
+    NA_character_, 
+    epic_celltype
+  ))
+
+
+# Next, we'll define a vector of colors to use for plots which ensures that shared cell types have the same colors
+# hex codes are (mostly) from Okabe-Ito
+color_map <- c(
+  # shared cell types
+  "Bcells" = "#E69F00", 
+  "CD4_Tcells" = "#56B4E9", 
+  "CD8_Tcells" = "#009E73", 
+  "NKcells" =  "#F0E442",
+  # reference-specific cell types
+  "CAFs" = "#0072B2",
+  "Endothelial" = "#D55E00",
+  "Macrophages" = "#CC79A7",
+  "Monocytes" = "grey30", 
+  "Neutrophils" = "grey60"
+)  
+celltype_levels <- c(names(color_map))
+  
+  
+tref_barplot <- make_epic_celltype_barplot(
+  epic_otherna_df |> dplyr::filter(reference == "TRef"),
+  xvar = sample_id, 
+  yvar = fraction, 
+  facetvar = project_diagnosis,
+  fillvar = forcats::fct_relevel(epic_celltype, celltype_levels),
+  plot_title = "TRef reference", 
+  color_map = color_map
+) 
+  
+bref_barplot <- make_epic_celltype_barplot(
+  epic_otherna_df |> dplyr::filter(reference == "BRef"),
+  xvar = sample_id, 
+  yvar = fraction, 
+  facetvar = project_diagnosis,
+  fillvar = forcats::fct_relevel(epic_celltype, celltype_levels),
+  plot_title = "BRef reference", 
+  color_map = color_map
+) 
+
+tref_barplot / bref_barplot
+```
+
+Generally speaking, `TRef` identifies more cell types compared to `BRef`, although whether it _should_ be classifying all those cells is an open question to bear in mind!
+
+Let's compare the proportions of shared cell types inferred between references a bit closely with scatterplots:
+
+```{r fig.width = 12, fig.height = 14}
+shared_celltypes <- c("Bcells", "CD4_Tcells", "CD8_Tcells", "NKcells", "otherCells")
+
+
+compare_reference_celltype <- function(celltype, epic_df) {
```

Comment:
I don't love these plots... I think I would rather just see a sina plot where you show the proportions of each cell type on the y-axis and the reference on the x-axis.